src/pages/index.js

```diff
@@ -148,23 +233,123 @@ function handleEscapeKey(event) {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // TODO Call setButtonText instead
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
 }
 
+// TODO - implement loading text for all other form submissions
+
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
+  const submitBtn = evt.submitter;
 
-  console.log(cardNameInput);
-  console.log(cardLinkInput);
+  setButtonText(submitBtn, true, "save", "saving...");
 
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardEl = getCardElement(inputValues);
-  cardsList.prepend(cardEl);
-  evt.target.reset(settings);
-  disableButton(cardSubmitBtn, settings);
-  closeModal(cardModal);
+  api
+    .addCard(inputValues)
+    .then((res) => {
+      // renderCard(res, "prepend"); // optional
+
+      const cardEl = getCardElement(res);
+      cardsList.prepend(cardEl);
+      evt.target.reset(settings);
+      disableButton(cardSubmitBtn, settings);
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "save", "saving...");
+    });
+}
+
+// TODO - FINISH avatar submission handler
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const newAvatarUrl = avatarInput.value;
+
+  // TODO - prevent behavior
+  // TODO - Call api.editAvatarUserInfo
+
+  if (!newAvatarUrl) {
+    return;
+  }
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editAvatarInfo(newAvatarUrl)
+    .then((data) => {
+      console.log(data.avatar);
+      // make this work- add the src request
+      const profileAvatar = document.querySelector(".profile__avatar");
+      if (profileAvatar) {
+        profileAvatar.src = data.avatar;
+      }
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "saving...");
+    });
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+      const selectedCardElement = document.querySelector(
+        `#card-${selectedCardId}`
+      );
+      if (selectedCardElement) {
+        selectedCardElement.remove();
+      }
+      // TODO //
+      // REMOVE THE CARD FROM THE DOM
+      // CLOSE THE MODAL
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Delete", "Deleting...");
+    });
+}
+
+function handleDeleteCard(cardElement, id) {
+  selectedCard = cardElement;
+  selectedCardId = id;
+
+  const openModals = document.querySelectorAll(".modal_opened");
```

Comment:
```js
  const openModals = document.querySelectorAll(".modal_opened");
  openModals.forEach((modal) => closeModal(modal));
```

It's not needed when you open any modal. The modals are already closed if you can open any other modals


```js
function handleDeleteCard(cardElement, id) {
  selectedCard = cardElement;
  selectedCardId = id;
  openModal(deleteModal);
}

```

Reply:
My remark hasn't been corrected